scipy/stats/tests/test_morestats.py

```diff
@@ -1636,34 +1629,26 @@ def test_exact_basic(self):
             assert_equal(sum(pmf1), 1)
             assert_array_almost_equal(pmf1, pmf2)
 
-        # make sure invalid calculation of z due to division by 0 is skipped
-        # warning would cause a test failure
-        msg = "Exact p-value calculation does not work if there are zeros"
-        with pytest.warns(UserWarning, match=msg):
-            w, p = stats.wilcoxon(np.zeros(5), method="exact")
-        assert_equal((w, p), (0.0, 1.0))
```

Comment:
This didn't looked like it belonged in `test_exact_basic`. I don't really think it needs to be tested at all, since it does not seem like such a bad thing for the user to see that calculations break down when their data is degenerate, but if it's important, we can add it back as a separate test.

Reply:
There was the problem that the z-statistic was always computed, even if it is not needed. That caused a warning (division by zero, not the warning that there are zeros in d) unrelated to the calculation of the exact p-value. This is fixed now and the idea of the test was to ensure that the division by zero warning does not reappear.

Reply:
OK. If that is the intent, let's make it a separate test with a comment about that. As-is, the test looks like it's only to test the warning about exact p-value calculation, and it looks misplaced in `test_exact_basic`.